apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,182 @@
+      uid: uid,
+      course_instance_id: course_instance_id,
+    });
+    if (!user) {
+      throw error.make(400, `User ${uid} is not enrolled in this course instance.`);
+    }
+
+    return { user_id: user.user_id, group_id: null };
+  } else {
+    throw error.make(400, 'Student UID or Group Name is required.');
+  }
+}
+
+async function selectUserEnrolledInCourseInstance({ uid, course_instance_id }) {
+  const user = await selectUserByUid(uid);
+  if (!user) return null;
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id: course_instance_id,
+  });
+  if (!enrollment) return null;
+
+  return user;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+    });
+    res.locals.policies = result.rows;
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
+
+    if (req.body.__action === 'add_new_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+
+      
+      await runInTransactionAsync(async () => {
+        const inserted = await sqldb.queryAsync(sql.insert_assessment_access_policy, {
+          assessment_id: res.locals.assessment.id,
+          created_by: res.locals.authn_user.user_id,
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
+          group_name: req.body.group_name || null,
+          note: req.body.note || null,
+          start_date: new Date(req.body.start_date),
+          group_id: group_id || null,
+          user_id: user_id || null,
+        });
+        await insertAuditLog({
+          authn_user_id: res.locals.authn_user.user_id,
+          table_name: 'assessment_access_policies',
+          action: 'insert',
+          institution_id: res.locals.institution.id,
+          course_id: res.locals.course.id,
+          course_instance_id: res.locals.course_instance.id,
+          new_state: JSON.stringify(inserted.rows),
+          row_id: inserted.rows[0].id,
+        });
+      });
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_override') {
+      
+      await runInTransactionAsync(async () => {
+        const deletedAccessPolicy = await sqldb.queryAsync(sql.delete_assessment_access_policy, {
```

Comment:
Again, let's bake in some assumptions:

```suggestion
        const deletedAccessPolicy = await sqldb.queryZeroOrOneRowAsync(sql.delete_assessment_access_policy, {
```

Reply:
done in 45e18b539332b0b378564286a14cee723eb1e01f